orcid_service/views.py

```diff
@@ -282,6 +381,58 @@ def update_profile(orcid_id, data=None):
             # per PEP-0249 a transaction is always in progress
             session.commit()
 
+def update_profile_local(orcid_id, data=None, force=False):
+    """Update local db with ORCID profile"""
+
+    data = json.loads(data)
+
+    with current_app.session_scope() as session:
+        profile = session.query(Profile).filter_by(orcid_id=orcid_id).first()
+        if not profile:
+            logging.error('ORCID profile {} does not exist; creating'.format(orcid_id))
+            profile = Profile(orcid_id=orcid_id, created=datetime.utcnow())
+            force = True
+        # data assumed to come from ORCID API /works endpoint
+        if data:
+            # convert milliseconds since epoch to seconds since epoch
+            last_modified = data['activities-summary']['last-modified-date']['value']
+            last_modified /= 1000.
+            if force or (profile.updated < datetime.fromtimestamp(last_modified)):
+                works = data['activities-summary']['works']['group']
+                new_recs = {}
+                update_recs = {}
+                orcid_recs = []
+                try:
+                    current_recs = profile.bibcode.keys()
+                except:
+                    current_recs = []
+                for work in works:
+                    try:
+                        id0, rec = find_record(work)
+                    except:
+                        continue
+                    if id0 not in current_recs:
+                        new_recs.update(rec)
```

Comment:
The original intent was to ignore the existing records that haven't been updated in the time since the profile was updated. rec[id0]['updated'] is the update date on a single work in the ORCID profile, so the block below only selects and updates the metadata on records that have been updated on the ORCID side since our local profile was last touched (profile.updated), and ignores untouched existing local records. I guess there's not any harm in just automatically updating the metadata for all records, though, which is what a straight `else` would give you.